File: Structural-Patterns/Decorator/Java/Interfaces/CharacterDecorator.java
package Interfaces;

import java.util.List;

public abstract class CharacterDecorator implements ICharacter {
    protected ICharacter character;

    public CharacterDecorator(ICharacter character){
        this.character = character;
    }

    @Override
    public String getName() {
        return character.getName();
    }

    @Override
    public int getAttackPower() {
        return character.getAttackPower();
    }

    @Override
    public int getDefense() {
        return character.getDefense();
    }

    @Override
    public String getDescription() {
        return character.getDescription();
    }

    @Override
    public List<String> getAbilities() {
        return character.getAbilities();
    }
}

File: Structural-Patterns/Decorator/C#/Decorator/Interfaces/CharacterDecorator.cs
public abstract class CharacterDecorator : ICharacter {
    protected ICharacter character;

    public CharacterDecorator(ICharacter character){
        this.character = character;
    }

    public virtual String GetName() {
        return character.GetName();
    }

    public virtual int GetAttackPower() {
        return character.GetAttackPower();
    }

    public virtual int GetDefense() {
        return character.GetDefense();
    }

    public virtual String GetDescription() {
        return character.GetDescription();
    }

    public virtual List<String> GetAbilities() {
        return character.GetAbilities();
    }
}
